src/infix-abstract.ts

```diff
@@ -0,0 +1,69 @@
+
+const SUBSCRIPT_DIGITS = new Map(Object.entries({
+  "0": "₀",
+  "1": "₁",
+  "2": "₂",
+  "3": "₃",
+  "4": "₄",
+  "5": "₅",
+  "6": "₆",
+  "7": "₇",
+  "8": "₈",
+  "9": "₉",
+  "-": "₋"
+}));
+
+
+import { Notation } from "./notation";
+import Decimal from "break_infinity.js";
+
+export abstract class AbstractInfixNotation extends Notation {
+  public get name(): string {
+    return "Abstract Infix";
+  }
+
+  public formatDecimal(value: Decimal, places: number): string {
+    const p = Math.max(places, this.groupDigits-1);
+//     const offset = value.exponent % this.groupDigits;
+//     let exp = value.exponent - offset;
+    let exp = value.exponent;
+//     let mant = value.mantissa * Math.pow(10, offset);
+    let mant = value.mantissa;
+
+    let str = "";
+    // Loop through digits, from first to next-to-last
+    for (let minExp = exp - p; exp > minExp; exp--) {
```

Comment:
Codacy found an issue: [Identifier 'minExp' is never reassigned; use 'const' instead of 'let'.](https://app.codacy.com/gh/antimatter-dimensions/notations/pullRequest?prid=6218257)